src/endpoint/s3/ops/s3_post_object_restore.js

```diff
@@ -1,32 +1,44 @@
 /* Copyright (C) 2023 NooBaa */
 'use strict';
 
+const { S3Error } = require('../s3_errors');
 const s3_utils = require('../s3_utils');
 
 /**
  * https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html
  */
 async function post_object_restore(req, res) {
     const encryption = s3_utils.parse_encryption(req);
-    const days = req.body?.RestoreRequest?.Days ? parseInt(req.body.RestoreRequest.Days?.[0], 10) : undefined;
+
+    if (!req.body?.RestoreRequest?.Days?.[0]) {
+        throw new S3Error(S3Error.MalformedXML);
+    }
+
+    const days = parseInt(req.body.RestoreRequest.Days[0], 10);
+    if (days < 1) {
+        throw new S3Error(S3Error.InvalidArgument);
+    }
+
     const params = {
         bucket: req.params.bucket,
         key: req.params.key,
         version_id: req.query.versionId,
-        days: days,
+        days,
         encryption,
     };
 
-
-    await req.object_sdk.restore_object(params);
-    res.statusCode = 200;
+    const success = await req.object_sdk.restore_object(params);
+    if (success) {
```

Comment:
perhaps rename this const to `accepted` to match the http statusCode 202 meaning.